starknet_py/cairo/v1/type_parser.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from typing import Dict, Union
+
+from starknet_py.abi.v1.parser_transformer import parse
+from starknet_py.cairo.data_types import CairoType, EnumType, StructType, TypeIdentifier
+
+
+class UnknownCairoTypeError(ValueError):
+    """
+    Error thrown when TypeParser finds type that was not declared prior to parsing.
+    """
+
+    type_name: str
+
+    def __init__(self, type_name: str):
+        super().__init__(f"Type '{type_name}' is not defined")
+        self.type_name = type_name
+
+
+class TypeParser:
+    """
+    Low level utility class for parsing Cairo types that can be used in external methods.
+    """
+
+    defined_types: Dict[str, Union[StructType, EnumType]]
+
+    def __init__(self, defined_types: Dict[str, Union[StructType, EnumType]]):
+        """
+        TypeParser constructor.
+
+        :param defined_types: dictionary containing all defined types. For now, they can only be structures.
+        """
+        self.defined_types = defined_types
+        for name, defined_type in defined_types.items():
+            if name != defined_type.name:
+                raise ValueError(
+                    f"Keys must match name of type, '{name}' != '{defined_type.name}'."
+                )
```

Comment:
Wasn't this supposed to be renamed?

Reply:
which `this`? I've renamed `structure` -> `defined_type`

Reply:
Okay, right. It's fine.